utils/gooey_buttons.lua

```diff
@@ -65,6 +65,10 @@ function M.plus_button(button)
 	end
 end
 
+function M.set_long_pressed_time(button, time)
+	button.long_pressed_time = time
+end
+
```

Comment:
Don't think we need this. We can simply call `gooey.button(node).set_long_pressed_time(time)` directly

Reply:
I clearly do not understand how lua "class" equivalents work.

Reply:
It returns a `table` (kind of it's own module) that have a function stored on the key `set_long_pressed_time`, was confusing as hell for me too after only have been using Python before starting with Defold.

Reply:
Yeah, I get the concept, I just don't understand how to spot it or what the standards or keywords are for lua.

Like, in C# you'd have something like:
```
namespace Gooey
{
    public class Button
    {
        private double long_pressed_time = 0.0;

        public void set_long_pressed_time(double t)
        {
            long_pressed_time  = t;
        }

// well ACTUALLY this would be a property with a getter/setter but wanted it to match
    }
}
```

and then your usage code would be like:
```
    var myButton = new Gooey.Button();
    ...
    myButton.set_long_pressed_time(0.5);
```

In lua I don't know how to identify when this sort of thing is happening.

Reply:
Reading the source code I guess 🙃 [gooey.button](https://github.com/britzl/gooey/blob/master/gooey/internal/button.lua), but if you don't really follow it (even into core.instance) you will have to assume it does so.

I think the API mention that this is how you set the settings to right? If not I should add it to gooeys API

Reply:
I was considering anything under "internal" to not be part of the public API.